q2_vizard/tests/test_lineplot.py

```diff
@@ -0,0 +1,192 @@
+            line_elements = \
+                driver.find_elements(By.CSS_SELECTOR, 'g.mark-line.role-mark')
+            md_df = self.md.to_dataframe()
+
+            if facet_measure:
+                exp_lines = len(md_df[facet_measure].unique())
+            else:
+                exp_lines = 1
+
+            self.assertEqual(len(line_elements), exp_lines)
+
+            # test that the correct number of scatter marks are present
+            # and a mark is where we expect it to be
+            # for some reason this viz renders multiple groups
+            # of the same marks, so just selecting the first one
+            first_mark_container = \
+                driver.find_element(By.CSS_SELECTOR,
+                                    'g.mark-symbol.role-mark')
+            mark_elements = \
+                first_mark_container.find_elements(By.CSS_SELECTOR, 'path')
+
+            self.assertEqual(exp_marks_len, len(mark_elements))
+
+            mark_element_0 = mark_elements[0]
+            mark_class = mark_element_0.get_attribute('class')
+            mark_x = mark_element_0.get_attribute('data-x')
+            mark_y = mark_element_0.get_attribute('data-y')
+
+            self.assertEqual(mark_class, exp_mark_class)
+            self.assertEqual(mark_x, exp_x_mark)
+            self.assertEqual(mark_y, exp_y_mark)
+
+    # run selenium checks with a chrome driver
+    def test_lineplot_chrome(self):
+        chrome_options = ChromeOptions()
+        chrome_options.add_argument('-headless')
+
+        with webdriver.Chrome(options=chrome_options) as driver:
+            for (x_measure, y_measure, facet_measure, replicate_method,
+                 exp_subtitle, exp_legend, exp_marks_len, exp_mark_class,
+                 exp_x_mark, exp_y_mark) in self.test_cases:
+
+                with self.subTest(
+                    x_measure=x_measure, y_measure=y_measure,
+                    facet_measure=facet_measure,
+                    replicate_method=replicate_method,
+                    exp_subtitle=exp_subtitle, exp_legend=exp_legend,
+                    exp_marks_len=exp_marks_len, exp_mark_class=exp_mark_class,
+                    exp_x_mark=exp_x_mark, exp_y_mark=exp_y_mark
+                ):
+
+                    self._selenium_lineplot_test(
+                        driver, x_measure, y_measure, facet_measure,
+                        replicate_method, exp_subtitle, exp_legend,
+                        exp_marks_len, exp_mark_class, exp_x_mark, exp_y_mark)
+
+    def test_lineplot_firefox(self):
+        firefox_options = FirefoxOptions()
+        firefox_options.add_argument('-headless')
+
+        with webdriver.Firefox(options=firefox_options) as driver:
+            for (x_measure, y_measure, facet_measure, replicate_method,
+                 exp_subtitle, exp_legend, exp_marks_len, exp_mark_class,
+                 exp_x_mark, exp_y_mark) in self.test_cases:
```

Comment:
I would throw an `assert len(self.test_cases) > 0` somewhere to assuage the reader, as I've been burned by the for-loop in a test before.